src/types.ts

```diff
@@ -1,37 +1,37 @@
 export type UnbindFn = () => void;
 
-type ExtractEventTypeFromHandler<MaybeFn extends unknown> = MaybeFn extends (
-  this: any,
-  event: infer MaybeEvent,
-) => any
-  ? MaybeEvent extends Event
-    ? MaybeEvent
-    : Event
+type AnyFunction = (...args: any[]) => any;
+
+export type InferEventType<TTarget> = TTarget extends {
+  addEventListener(type: infer P, ...args: any): void;
+  addEventListener(type: infer P2, ...args: any): void;
```

Comment:
What is `addEventListener(type: infer P2, ...args: any): void;` (and `P2`) for?

Reply:
Added comments to the source code that explain it